pages/_app.tsx

```diff
@@ -1,10 +1,68 @@
 import CssBaseline from '@material-ui/core/CssBaseline'
-import { ThemeProvider } from '@material-ui/core/styles'
+import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
 import App from 'next/app'
 import Head from 'next/head'
-import React from 'react'
+import React, { useState, createContext } from 'react'
+import { red, blue, deepPurple } from '@material-ui/core/colors'
+import { Color, PaletteType } from '@material-ui/core'
+import { NextComponentType, NextPageContext } from 'next'
 
-import theme from '../components/theme'
+interface AppContextStructure {
+	themeType: 'dark' | 'light'
+	setThemeType: React.Dispatch<React.SetStateAction<PaletteType>>
+	themePrimary: Color
+	setPrimary: React.Dispatch<React.SetStateAction<Color>>
+	themeSecondary: Color
+	setSecondary: React.Dispatch<React.SetStateAction<Color>>
+}
+
+interface Props {
+	component: NextComponentType<NextPageContext, any, {}>
+	pageProps: any
+}
+
+export const AppContext = createContext<Partial<AppContextStructure>>({})
```

Comment:
I'd prefer it if the Context component was it's own thing, rather than part of the _app page. Could you extract this into it's own module in components/ and then just import it here?